be/src/exprs/agg/array_agg.h

```diff
@@ -226,7 +238,14 @@ class ArrayAggAggregateFunctionV2
             // release order-by columns early
             order_by_columns.clear();
             state_impl.release_order_by_columns();
-            DCHECK(ctx->state()->cancelled_ref() || st.ok());
+            if (UNLIKELY(ctx->state()->cancelled_ref())) {
```

Comment:
I think we shouldn't check the cancel status in aggregate function internal.

Reply:
this is effected by L237, as the sort need to know the cancel statues. if do know it in time, the cancel process would be long, instead of stopping as soon as possible.
`Status st = sort_and_tie_columns(ctx->state()->cancelled_ref(), order_by_columns, sort_desc, &perm);`